arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -210,6 +210,9 @@ public VectorHolder read(VectorHolder reuse, int numValsToRead) {
   }
 
   private void allocateFieldVector(boolean dictionaryEncodedVector) {
+    if (vec != null) {
+      vec.close();
+    }
```

Comment:
This is the fix for memory leaks when reading parquet files containing interleaving plain/dictionary pages.
`VectorizedArrowReader.read` allocates arrow vectors when encoding of pages in currently reading row group changes, it does not close the previously allocated vector before allocating new vector, which causes memory leaks.